qiskit/transpiler/passes/layout/csp_layout.py

```diff
@@ -144,5 +144,7 @@ def constraint(control, target):
         else:
             stop_reason = 'solution found'
             self.property_set['layout'] = Layout({v: qubits[k] for k, v in solution.items()})
+            for reg in dag.qregs.values():
+                self.property_set['layout'].add_register(reg)
```

Comment:
Wouldn't be better if `add_register` runs at construction time? otherwise, a layout can be "inconsistent", I think. `from_dict` could take care extending `Layout. _regs`